### DISCUSSION TITLE
Hardhat documentation update

### DISCUSSION THREAD

**Discussion by rashidmya** [NomicFoundation/hardhat#5178] [Ideas]
I think documentation for tools and plugins should include more information for classes and functions like properties, methods, what they return, usage, etc..

Examples: 
- https://docs.ethers.org/v6/single-page/#api_contract__ContractFactory
- https://nodejs.org/docs/latest/api/stream.html#streampipelinestreams-options
- https://react.dev/reference/react/createContext

**Comment by kanej** [comment]
I think that is a good idea. We wouldn't want to detract from the guide documentation focused on achieving a task, but adding them in some sort of reference section makes sense.

Are there particular interactions with the `hre` or a plugin were the lack of class/type info has been a particular problem?
